src/applications/disability-benefits/2346/containers/OrderTable.jsx

```diff
@@ -0,0 +1,186 @@
+import ErrorableCheckbox from '@department-of-veterans-affairs/formation-react/ErrorableCheckbox';
+import React from 'react';
+import { connect } from 'react-redux';
+import { focusElement } from '../../../../platform/utilities/ui';
+import { getDLCData, updateDLCData } from '../actions';
+
+class OrderHistory extends React.Component {
+  // NOTE: Using React state for checkboxes -@maharielrosario at 1/8/2020, 7:47:21 PM
+  // When discussed with other devs, the concensus was to have a stateful
+  // component to house our checkboxes' state. But all other state is in Redux.
+  // Is there an easy way to house individual checkbox state in Redux?
+
+  constructor(props) {
+    super(props);
+    this.createDlcUIData = this.createDlcUIData.bind(this);
+  }
+
+  componentDidMount() {
+    focusElement('.va-nav-breadcrumbs-list');
+    this.props.getDLCData();
+  }
+
+  createDlcUIData(initDataObjs) {
+    let UIDataObjs = initDataObjs;
+
+    (function removeNonUIProps() {
```

Comment:
Any reason we're making an IIFE here?

Reply:
So you could do something like this instead:
```jsx
const {
  data: {
    key1YouNeed,
    key2YouNeed,
  },
} = this.props;
```

Reply:
Alternatively:
```jsx
const { data } = this.props;

// Derive data properties.
const key1YouNeed = data?.key1YouNeed;
const key2YouNeed = data?.key2YouNeed;
```

Reply:
@kelsonic So this is essentially sculpting my object to have the keys I want?

  `const {
      UIDataObjs: {
        dateProcessed,
        product,
        id,
        orderIntakeMethod
      }
    }`

Reply:
Yupp! You can omit the `for` loop and do what I suggested above if you'd like 🙂This is called [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), it's pretty handy!